pkg/iac/terraform/state/backend/http_reader.go

```diff
@@ -31,6 +33,16 @@ func NewHTTPReader(rawURL string, opts *Options) (*HTTPBackend, error) {
 		return nil, err
 	}
 
+	buf := new(bytes.Buffer)
+	_, err = buf.ReadFrom(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode < 200 || res.StatusCode >= 400 {
+		return nil, fmt.Errorf("error in backend HTTP(s): non-200 OK status code: %s body: \"%s\"", res.Status, buf.String())
```

Comment:
I'm a little confused with this error message. 

```
DEBU[0001] Closing providers
DEBU[0001] Closing provider key=aws
DEBU[0001] Closing gRPC client alias=eu-west-3
error in backend HTTP(s): non-200 OK status code: 401 Unauthorized body: "{"message":"401 Unauthorized"}"
```

Maybe we can display something simpler:

```go
return nil, fmt.Errorf("Error requesting HTTP(s) backend state: \"%s\"", buf.String())

// Error requesting HTTP(s) backend state: "401 Unauthorized"
```

Reply:
I followed the code snippet from `pkg/remote/error/errors.go` : 

```
error in supplier github_branch_protection: non-200 OK status code: 401 Unauthorized body: "{\"message\":\"Bad credentials\",\"documentation_url\":\"https://docs.github.com/graphql\"}"
```

Isn't it more productive for the user to get to response body as well? Returning a 400 error without the body will prevent the user from understand the issue.

Reply:
The part where it says `401 Unauthorized` is the body. It is the first part that I'm not okay with (e.g. `non-200 OK status code: 401), I don't understand a thing ... Either we output only `error ...: status code: 401, body: "<message>"` or we only output the body.

Reply:
@eliecharra WDYT ? What could be a good error message ?

Reply:
I think it's enough here if we keep a simple message only with the http status code : `Error requesting HTTP(s) backend state: HTTP code 4xx`

For me the body is related to debug purpose, maybe we can add a TRACE log with the response body.

Reply:
That's way better I agree with Elie.